### PR TITLE
IA-3955: removing workspaceId from disk v2 route

### PR DESCRIPTION
**Pull request by lucymcnatt**
workspaceId isn't required in the initial logic for the deleteDisk v2 route, so this removes it from the path

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterp.scala [hunk 1]
```diff
@@ -120,3 +123,10 @@ case class DiskCannotBeDeletedAttachedException(id: DiskId, workspaceId: Workspa
       StatusCodes.Conflict,
       traceId = Some(traceId)
     )
+
+case class DiskWithoutWorkspaceException(id: DiskId, traceId: TraceId)
```
Comment (n8landolt) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterp.scala:127]: I assume this scenario should never happen?

Is the reason for this so a user can't delete a disk they're not allowed to?

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterpSpec.scala [hunk 2]
```diff
@@ -192,9 +192,36 @@ class DiskV2ServiceInterpSpec extends AnyFlatSpec with LeonardoTestSuite with Te
...
         )
       )
-      err <- diskV2Service.deleteDisk(userInfo, workspaceId, disk.id).attempt
+      err <- diskV2Service.deleteDisk(userInfo, disk.id).attempt
     } yield err shouldBe Left(DiskCannotBeDeletedAttachedException(disk.id, workspaceId, ctx.traceId))
 
     res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
   }
+
+  it should "fail to delete a disk if it has no workspaceId" in isolatedDbTest {
```
Comment (n8landolt) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterpSpec.scala:201]: Is it possible to create a disk without a workspaceId?

Comment (n8landolt) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterpSpec.scala:201]: Adding in this logic concerns me, it seems to be just another layer of obscure functionality that restricts user control - if this scenario happens, manual intervention is required.

Comment (lucymcnatt) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterpSpec.scala:201]: with an Azure disk no, but with a GCP runtime yes. The workspaceId column in the PERSISTENT_DISK table is an option, so in order to get the workspaceId I used that fromOption logic. I don't think this exception should happen very often/at all, but thought it might be useful when we switch over to using the v2 routes for GCP as well.

Comment (n8landolt) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterpSpec.scala:201]: Ahh it seems like it's used by a dependency - WSM.
In Leo it seems like workspaceId is strictly for logging, but once we make the WSM delete request it needs the workspaceId

File: env/local.env [hunk 3]
```diff
@@ -101,3 +101,6 @@ LEO_EXECUTION_MODE=backLeo
 
 # Akka logging
 AKKA_LOG_LEVEL=INFO
+
+# Java
+JAVA_HOME=java.home
```
Comment (n8landolt) [inline env/local.env:106]: Is this change intentional?

File: http/src/main/resources/leo.conf [hunk 4]
```diff
@@ -123,7 +123,7 @@ akka {
         },
         {
           # Default trust store
-          path: ${JAVA_HOME}/lib/security/cacerts
+          path: ${?JAVA_HOME}/lib/security/cacerts
```
Comment (n8landolt) [inline http/src/main/resources/leo.conf:126]: Same with this - is this intentional?

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterpSpec.scala (hunk 2 above)
Comment (lucymcnatt) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterpSpec.scala:201]: yup yeah that's why its needed in the backend call, every call to WSM for azure disks, runtimes etc requires a workspaceId

File: http/src/main/resources/leo.conf (hunk 4 above)
Comment (lucymcnatt) [reply http/src/main/resources/leo.conf:126]: ohh yeah, I can get rid of them cuz they're out of the scope of this change, but leo tests don't run locally without JAVA_HOME being defined

Comment (n8landolt) [reply http/src/main/resources/leo.conf:126]: I'm okay with the java changes going in - it doesn't affect anything other than the local tests.

**Review by n8landolt — APPROVED** [review]

**Comment by lucymcnatt** [comment]
jenkins retest
